app/tests/spec/lib/fxa-client.js

```diff
@@ -646,6 +646,180 @@ define(function (require, exports, module) {
+
+        relier.set('service', NON_SYNC_SERVICE);
+        sinon.stub(relier, 'wantsKeys').callsFake(() => false);
+
+        return client.sessionReauth(sessionToken, email, password, relier)
+          .then(function (sessionData) {
+            assert.isTrue(realClient.sessionReauth.calledWith(sessionToken, trim(email), password, {
+              keys: false,
+              reason: SignInReasons.SIGN_IN,
+              redirectTo: REDIRECT_TO,
+              service: NON_SYNC_SERVICE
+            }));
+
+            // These should not be returned by default
+            assert.isFalse('unwrapBKey' in sessionData);
+            assert.isFalse('keyFetchToken' in sessionData);
+          });
+      });
+
+      it('passes along an optional `reason`', function () {
+        sinon.stub(relier, 'wantsKeys').callsFake(function () {
+          return true;
+        });
+
+        sinon.stub(realClient, 'sessionReauth').callsFake(function () {
+          return Promise.resolve({});
+        });
+
+        return client.sessionReauth(sessionToken, email, password, relier, { reason: SignInReasons.PASSWORD_CHANGE })
+          .then(function () {
+            assert.isTrue(realClient.sessionReauth.calledWith(sessionToken, trim(email), password, {
+              keys: true,
+              reason: SignInReasons.PASSWORD_CHANGE,
+              redirectTo: REDIRECT_TO,
+              service: SYNC_SERVICE
+            }));
+          });
+      });
+
+      it('passes along an optional `resume`', function () {
+        sinon.stub(realClient, 'sessionReauth').callsFake(function () {
+          return Promise.resolve({});
+        });
+
+        return client.sessionReauth(sessionToken, email, password, relier, { resume: 'resume token' })
+          .then(function () {
+            assert.isTrue(realClient.sessionReauth.calledWith(sessionToken, trim(email), password, {
+              keys: false,
+              reason: SignInReasons.SIGN_IN,
+              redirectTo: REDIRECT_TO,
+              resume: 'resume token',
+              service: SYNC_SERVICE
+            }));
+          });
+      });
+
+      it('passes along an optional `metricsContext`', function () {
+        sinon.stub(realClient, 'sessionReauth').callsFake(function () {
+          return Promise.resolve({});
+        });
+
+        relier.set('service', NON_SYNC_SERVICE);
+
+        return client.sessionReauth(sessionToken, email, password, relier, {
+          metricsContext: { foo: 'bar' }
+        }).then(function () {
+          assert.isTrue(realClient.sessionReauth.calledWith(sessionToken, trim(email), password), {
+            keys: false,
+            metricsContext: { foo: 'bar' },
+            reason: SignInReasons.SIGN_IN,
+            service: NON_SYNC_SERVICE
+          });
+        });
+      });
+
+      it('passes along an optional `skipCaseError`', () => {
+        sinon.stub(realClient, 'sessionReauth').callsFake(() => Promise.resolve({}));
```

Comment:
repeated several times, as other comment suggests, can this be moved out of tests to beforeEach or something